sql: add crdb_internal.node_runtime_info

tbg opened the pull request:
Inspired by https://github.com/cockroachdb/cockroach/pull/17686, for which
this was useful (to obtain the AdminURL's port). I'm not sure if there are
security implications (i.e. should this be limited to root if it isn't
already?), but this seemed handy to have.

```
root@:26257/> SELECT * FROM crdb_internal.node_runtime_info;
+---------+----------+---------------------------------------------------+
| node_id |  field   |                       value                       |
+---------+----------+---------------------------------------------------+
|       1 | AdminURL | http://macts-2.local:50558                        |
|       1 | PGURL    | postgresql://@macts-2.local:26257?sslmode=disable |
+---------+----------+---------------------------------------------------+
(2 rows)
```

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/17762)
<!-- Reviewable:end -->

bdarnell commented:
I think I'd put hostname, port, and http_port here instead of the full urls. Let the caller assemble the URLs (which, for secure clusters, will entail knowledge of the client's cert location which may differ from the server's). 

I think I'd make this root-only (consider a DBaaS - you don't want everyone with sql access to be able to get details about your internal network). 

---

Review status: 0 of 4 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/17762#-:-KrwffGrtEimU0jP__zz:b-ua7fgm)*
<!-- Sent from Reviewable.io -->

tschottdorf commented:
PTAL! @knz We reserve the right to change `crdb_internal` as we please, right? Is there a reason to add it now?

---

Review status: 0 of 8 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/17762#-:-KsBwe-RZG1qmkOaLBCR:bdabh3n)*
<!-- Sent from Reviewable.io -->

tbg committed:
sql: add crdb_internal.node_runtime_info

Inspired by https://github.com/cockroachdb/cockroach/pull/17686, for which
this was useful (to obtain the AdminURL's port). I'm not sure if there are
security implications. Access to the new table is limited to the `root` user.

```
root@:26257/> select * from crdb_internal.node_runtime_info;
+---------+-----------+--------+------------------------------------------------------+
| node_id | component | field  |                        value                         |
+---------+-----------+--------+------------------------------------------------------+
|       1 | DB        | URL    | postgresql://root@macts-2.home:26257?sslmode=disable |
|       1 | DB        | Scheme | postgresql                                           |
|       1 | DB        | User   | root                                                 |
|       1 | DB        | Host   | macts-2.home                                         |
|       1 | DB        | Port   |                                                26257 |
|       1 | DB        | Path   |                                                      |
|       1 | UI        | URL    | http://macts-2.home:8080                             |
|       1 | UI        | Scheme | http                                                 |
|       1 | UI        | User   |                                                      |
|       1 | UI        | Host   | macts-2.home                                         |
|       1 | UI        | Port   |                                                 8080 |
|       1 | UI        | Path   |                                                      |
+---------+-----------+--------+------------------------------------------------------+
```

knz commented:
Oh yes thanks for the reminder. Let's merge this!

tschottdorf commented:
TFTRs!
